lib/charms/operator_libs_linux/v1/systemd.py

```diff
@@ -66,149 +71,146 @@
 class SystemdError(Exception):
     """Custom exception for SystemD related errors."""
 
-    pass
 
+def _systemctl(*args: str, check: bool = False) -> Optional[int]:
+    """Control a system service using systemctl.
 
-def _popen_kwargs():
-    return {
-        "stdout": subprocess.PIPE,
-        "stderr": subprocess.STDOUT,
-        "bufsize": 1,
-        "universal_newlines": True,
-        "encoding": "utf-8",
-    }
+    Args:
+        *args: Arguments to pass to systemctl.
+        check:
+            Check the output of the systemctl command. Default: False.
+
+    Raises:
+        SystemdError:
+            Raised if returncode != 0 and check is True.
+            None will be returned instead of a returncode.
```

Comment:
When will `None` be returned from this function? From looking at the code structure below, it looks like it can only return an `int` exit code *or* raise `SystemdError`. If so, we should tweak this docstring and change this function to return type `int` instead of `Optional[int]`.

Reply:
In my haste I said that `None` is returned if check=True and the returncode!=0, however, this is not true on second glance. I will go ahead and tweak the docstring.

Reply:
Revised in commit https://github.com/canonical/operator-libs-linux/pull/106/commits/c951f68b8fe18cbab6fc69858e838283063c1e48